scripts/run_smoke_tests.py

```
@@ -6459,6 +6459,460 @@ def verify_legacy_link_install_detection(tmp_root: Path) -> None:
+                "    sys.exit(0)",
+                "if args[:1] == ['ls']:",
+                "    sys.exit(0 if (root / 'node_modules').is_dir() else 1)",
+                "if args[:2] == ['run', 'build']:",
+                "    if os.environ.get('STUDENT_OS_DSH_FAIL_BUILD'): sys.exit(9)",
+                "    (root / 'dist').mkdir(exist_ok=True)",
+                "    (root / 'dist' / 'index.js').write_text('export default {}\\n', encoding='utf-8')",
+                "    (root / 'dist' / 'index.d.ts').write_text('export {}\\n', encoding='utf-8')",
+                "    sys.exit(0)",
+                "sys.exit(0)",
+            ]
+        )
```

Ruff (0.16.1)

[warning] Consider f-string instead of string join

Replace with f-string

(FLY002)